### PR TITLE
Remove AdditionalProperties dictionary interface 

### PR DESCRIPTION
**Pull request by chamons**
# Description
- Fixes the parts of https://github.com/Azure/autorest.csharp/issues/1232 that we are hashing out now.
- Remove AdditionalProperties's IDictionary interface and make the field public

# Checklist

To ensure a quick review and merge, please ensure:
- [x] The PR has a understandable title and description explaining the _why_ and _what_.
- [x] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [x] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Comment by chamons** [comment]
Hey @pakrym - Not sure what you meant by your "might need to change it, we have a real property" comment on the commit.

File: src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs [hunk 1]
```diff
@@ -78,7 +78,7 @@ public SchemaObjectType(ObjectSchema objectSchema, BuildContext context) : base(
                 SourceMemberMapping? memberMapping = _sourceTypeMapping?.GetForMember("$AdditionalProperties");
```
Comment (pakrym) [inline src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs:78]: We should use the real property name here (`AdditionalProperties`)

**Comment by pakrym** [comment]
I suspect you might need to fix some tests.

File: src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs (hunk 1 above)
Comment (pakrym) [reply src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs:78]: ^^^

>Hey @pakrym - Not sure what you meant by your "might need to change it, we have a real property" comment on the commit.

This.

**Comment by chamons** [comment]
Moving back to draft as a test failure showed me that it is incomplete.

File: src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs (hunk 1 above)
Comment (chamons) [reply src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs:78]: Discussed in chat and we decided against this.

**Review by pakrym — APPROVED** [review]

**Comment by chamons** [comment]
Don't land this until after the next next autorest bump.

**Comment by chamons** [comment]
Wait for next week.

**Comment by chamons** [comment]
Decided to land this week and not land bump PR until next.

**Pull request merged**